hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/discovery/ClientDiscoverySpiTest.java

```diff
@@ -374,6 +375,26 @@ public void testMulticastDiscoveryEnabledNoLocalhost() {
         HazelcastClient.newHazelcastClient(clientConfig);
     }
 
+    @Test
+    public void testCustomDiscovery_startCalledOnce() {
+        HazelcastInstance instance = Hazelcast.newHazelcastInstance();
+
+        ClientConfig config = new ClientConfig();
+        config.setProperty(ClusterProperty.DISCOVERY_SPI_ENABLED.getName(), "true");
+
+        final DiscoveryService discoveryService = mock(DiscoveryService.class);
+        when(discoveryService.discoverNodes()).thenReturn(Collections.singletonList(new SimpleDiscoveryNode(instance.getCluster().getLocalMember().getAddress())));
+        DiscoveryServiceProvider discoveryServiceProvider = arg0 -> discoveryService;
+        ClientNetworkConfig networkConfig = config.getNetworkConfig();
+        config.getConnectionStrategyConfig().getConnectionRetryConfig().setClusterConnectTimeoutMillis(0);
+        networkConfig.getDiscoveryConfig().addDiscoveryStrategyConfig(new DiscoveryStrategyConfig());
```

Comment:
Is it mandatory to add a discovery strategy?

Reply:
I just copied the test body from the other test. Let me remove it